lib/pbench/test/unit/server/test_datasets_contents.py

```diff
@@ -0,0 +1,278 @@
+        response = query_get_as("test", "metadata.log", HTTPStatus.FORBIDDEN)
+        assert response.json == {
+            "message": "User drb is not authorized to READ a resource owned by test with private access"
+        }
+
+    @pytest.mark.parametrize("key", ("", ".", "subdir1"))
+    def test_path_is_directory(self, query_get_as, monkeypatch, key):
+        def mock_find_entry(_s, _d: str, path: Optional[Path]):
+            file = path if path else Path(".")
+            return {
+                "children": {},
+                "details": CacheObject(
+                    file.name if key else "",
+                    file,
+                    None,
+                    None,
+                    None,
+                    CacheType.DIRECTORY,
+                ),
+            }
+
+        monkeypatch.setattr(CacheManager, "find_entry", mock_find_entry)
+        response = query_get_as("fio_2", key, HTTPStatus.OK)
+        name = "" if key == "." else key
+        assert response.json == {
+            "directories": [],
+            "files": [],
+            "name": name,
+            "type": "DIRECTORY",
+            "uri": f"https://localhost/api/v1/datasets/random_md5_string4/contents/{name}",
+        }
+
+    def test_not_a_file(self, query_get_as, monkeypatch):
+        def mock_find_entry(_s, _d: str, path: Optional[Path]):
+            raise BadDirpath("Nobody home")
+
+        monkeypatch.setattr(CacheManager, "find_entry", mock_find_entry)
+        response = query_get_as("fio_2", "subdir1/f1_sym", HTTPStatus.NOT_FOUND)
+        assert response.json == {"message": "Nobody home"}
+
+    def test_file_info(self, query_get_as, monkeypatch):
+        name = "f1.json"
+
+        def mock_find_entry(_s, _d: str, path: Optional[Path]):
+            return {
+                "details": CacheObject(path.name, path, None, None, 16, CacheType.FILE)
+            }
+
+        monkeypatch.setattr(CacheManager, "find_entry", mock_find_entry)
+        response = query_get_as("fio_2", name, HTTPStatus.OK)
+        assert response.status_code == HTTPStatus.OK
+        assert response.json == {
+            "name": name,
+            "size": 16,
+            "type": "FILE",
+            "uri": f"https://localhost/api/v1/datasets/random_md5_string4/inventory/{name}",
+        }
+
+    def test_link_info(self, query_get_as, monkeypatch):
+        name = "test/slink"
+
+        def mock_find_entry(_s, _d: str, path: Optional[Path]):
+            return {
+                "details": CacheObject(
+                    path.name,
+                    path,
+                    Path("test1"),
+                    CacheType.DIRECTORY,
+                    None,
+                    CacheType.SYMLINK,
+                )
+            }
+
+        monkeypatch.setattr(CacheManager, "find_entry", mock_find_entry)
+        response = query_get_as("fio_2", name, HTTPStatus.OK)
+        assert response.status_code == HTTPStatus.OK
```

Comment:
Isn't line 133 redundant?  (Doesn't `query_get_as()` do that check using the third parameter?)

Ditto line 225.